src/Bicep.Core/Registry/Providers/TypesV1Archive.cs

```diff
@@ -24,15 +26,18 @@ public static async Task<Stream> GenerateProviderTarStream(IFileSystem fileSyste
             var indexJson = await fileSystem.File.ReadAllTextAsync(indexJsonPath);
             await AddFileToTar(tarWriter, "index.json", indexJson);
 
-            /* TODO:
-               * read index.json
-               * figure out paths to other .json files that need to be included
-               * read other .json files, add them to tgz
-            */
+            HashSet<string> uniqueTypePaths = getAllUniqueTypePaths(indexJsonPath);
+
+            foreach (string path in uniqueTypePaths)
+            {
+                var typesJson = await fileSystem.File.ReadAllTextAsync(path);
+                await AddFileToTar(tarWriter, "types.json", typesJson);
+            }
```

Comment:
See my comment below about relative paths - I'd recommend returning relative paths from `getAllUniqueTypePaths`, because you'll need to pass this in instead of hard-coding "types.json" when calling `AddFileToTar()`

```suggestion
            var uniqueTypePaths = getAllUniqueTypePaths(indexJsonPath);

            foreach (var relativePath in uniqueTypePaths)
            {
                var absolutePath = Path.Combine(indexJsonParentPath, relativePath);
                var typesJson = await fileSystem.File.ReadAllTextAsync(absolutePath);
                await AddFileToTar(tarWriter, relativePath, typesJson);
            }
```